tests/overrides/test_data_parallel.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import MagicMock
+
+import pytest
+import torch
+
+from pytorch_lightning.overrides.data_parallel import LightningDistributedModule
+
+
+def test_lightning_distributed_module_methods():
+    """ Test that the LightningDistributedModule redirects .forward() to the LightningModule methods. """
+    pl_module = MagicMock()
+    dist_module = LightningDistributedModule(pl_module)
+
+    batch = torch.rand(5)
+    batch_idx = 3
+
+    pl_module.training = True
+    pl_module.testing = False
+    dist_module(batch, batch_idx)
+    pl_module.training_step.assert_called_with(batch, batch_idx)
+
+    pl_module.training = False
+    pl_module.testing = True
+    dist_module(batch, batch_idx)
+    pl_module.test_step.assert_called_with(batch, batch_idx)
+
+    pl_module.training = False
+    pl_module.testing = False
+    dist_module(batch, batch_idx)
+    pl_module.validation_step.assert_called_with(batch, batch_idx)
```

Comment:
```suggestion
    pl_module.training = True
    dist_module(batch, batch_idx)
    pl_module.training_step.assert_called_with(batch, batch_idx)

    pl_module.testing = True
    dist_module(batch, batch_idx)
    pl_module.test_step.assert_called_with(batch, batch_idx)

    pl_module.testing = False
    dist_module(batch, batch_idx)
    pl_module.validation_step.assert_called_with(batch, batch_idx)
```
after #5419 it is enum under hood

Reply:
reverted...